### PR TITLE
Add basic implementation to commit votes

### PR DESCRIPTION
**Pull request by ptare**
Parts of this implementation will have to change when prices can
actually be requested and when results can actually be computed.

I tried to keep any assumptions about how price requests would work out of this PR, to avoid any conflicts.

Issue #381

### CODE REVIEW

File: v1/contracts/Voting.sol [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * @notice Reveal a previously committed vote for `identifier` at `time`.
+     * @dev The revealed `price` and `salt` must match the latest `hash` that `commitVote()` was called with. Only the
+     * committer can reveal their vote.
+     */
+    function revealVote(bytes32 identifier, uint time, int price, int salt) external {
+        VoteInstance storage voteInstance = requests[identifier][time];
+        require(keccak256(abi.encode(price, salt)) == voteInstance.committedHashes[msg.sender],
+                "Committed hash doesn't match revealed price and salt");
+        voteInstance.committedHashes[msg.sender] = bytes32(0);
```
Comment (mrice32) [inline v1/contracts/Voting.sol]: nit: I think `delete voteInstance.committedHashes[msg.sender]` is a bit clearer.

File: v1/contracts/Voting.sol [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * @title Voting system for Oracle.
+ * @dev Handles receiving and resolving price requests via a commit-reveal voting scheme.
+ */
+contract Voting {
+
+    struct VoteInstance {
+        // Maps (voterAddress) to their committed hash.
+        // A bytes32 of `0` indicates no commit or a commit that was already revealed.
+        // TODO(ptare): Do we prefer to store an extra boolean instead?
```
Comment (mrice32) [inline v1/contracts/Voting.sol]: IMO, this is way better than storing an accompanying boolean.

File: v1/test/Voting.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  const account2 = accounts[1];
+  const account3 = accounts[2];
+  const account4 = accounts[3];
+
+  const getRandomInt = () => {
+    return web3.utils.toBN(web3.utils.randomHex(32));
+  };
+
+  it("One voter, one request", async function() {
+    const voting = await Voting.new();
```
Comment (mrice32) [inline v1/test/Voting.js]: I think we should add this contract to the migrations and use `.deployed()`. We can do so in a later PR, but just wanted to note that.

File: v1/test/Voting.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+    // Can't reveal before committing.
+    assert(await didContractThrow(voting.revealVote(identifier, time, price, salt)));
+    // Can commit a new hash.
+    await voting.commitVote(identifier, time, hash);
+
+    // Voters can alter their commits.
+    const newPrice = getRandomInt();
+    const newSalt = getRandomInt();
+    const newHash = web3.utils.soliditySha3(newPrice, newSalt);
```
Comment (mrice32) [inline v1/test/Voting.js]: Didn't know this function existed - nice!

File: v1/contracts/Voting.sol (hunk 1 above)
Comment (ptare) [reply v1/contracts/Voting.sol]: That's way better, thanks!

File: v1/contracts/Voting.sol (hunk 2 above)
Comment (ptare) [reply v1/contracts/Voting.sol]: Sounds good, removed this TODO.

File: v1/test/Voting.js (hunk 3 above)
Comment (ptare) [reply v1/test/Voting.js]: I'll do that in a follow up PR.

**Pull request merged**